### PR TITLE
Traversal conversion

### LINKED ISSUE #164: Support Partwise <-> Timewise conversion
Traversing a `MusicXML` score either in a partwise or timewise fashion has its benefits.

Let's enable conversion between `Partwise` and `Timewise` elements.

### PR DESCRIPTION
**Pull request by jsbean**
This PR is a proof-of-concept implementation of conversion between `Partwise` and `Timewise` structures.

Resolves #164.

### CODE REVIEW

**Comment by jsbean** [comment]
In this PR, I split up the attributes shared between the `Timewise` and `Partwise` `Measure` types into a new type: `MeasureAttributes`.

One can still access these attributes by way of the `@dynamicMemberLookup` by way of `KeyPath`s. Thus, `measure.number`, et al. exists as far as autocompletion is concerned.

This may be an interesting pattern to follow for structures with `attribute-group`s (e.g., `PrintStyleAlign`). Here, we get the advantages of nesting as well as top-level access (see #173).

**Comment by jsbean** [comment]
I think this works pretty nicely. Perhaps there is a slightly more performant way to do the juggling, but this works for now. I will block the merge for this until we all agree on the API.

File: Tests/MusicXMLTests/TraversalConversionTests.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                .attributes(attributes),
+                .note(note)
+            ]
+        )
+        let timewiseMeasure = Timewise.Measure(number: "1", parts: [timewisePart])
+        let timewise = Timewise(header: header, measures: [timewiseMeasure])
+        XCTAssertEqual(timewise.toPartwise(), partwise)
+        XCTAssertEqual(partwise.toTimewise(), timewise)
+        XCTAssertEqual(timewise.toPartwise().toTimewise(), timewise)
+        XCTAssertEqual(partwise.toTimewise().toPartwise(), partwise)
```
Comment (bwetherfield) [inline Tests/MusicXMLTests/TraversalConversionTests.swift:50]: 😮pretty cool!

File: Sources/MusicXML/Complex Types/Partwise/Partwise.Measure.swift [hunk 2]
```diff
@@ -68,42 +74,40 @@ extension Partwise {
...
+
+        internal init(attributes: MeasureAttributes, musicData: [MusicData]) {
+            self.attributes = attributes
+            self.musicData = musicData
+        }
+
+        /// - Returns: A measure attribute.
+        public subscript <T> (dynamicMember keyPath: KeyPath<MeasureAttributes, T>) -> T {
+            return attributes[keyPath: keyPath]
+        }
```
Comment (bwetherfield) [inline Sources/MusicXML/Complex Types/Partwise/Partwise.Measure.swift:96]: Aha! Gotcha

**Review by bwetherfield — APPROVED** [review]
This is awesome!

File: Sources/MusicXML/Complex Types/Partwise/Partwise.Measure.swift (hunk 2 above)
Comment (jsbean) [reply Sources/MusicXML/Complex Types/Partwise/Partwise.Measure.swift:96]: I was always wondering when someone would want to use this…

**Review by DJBen — APPROVED** [review]

**Pull request merged**